test/common/upstream/cluster_update_tracker_test.cc

```diff
@@ -31,59 +31,53 @@ TEST_F(ClusterUpdateTrackerTest, ClusterDoesNotExistAtConstructionTime) {
 
   ClusterUpdateTracker cluster_tracker(cm_, cluster_name_);
 
-  EXPECT_FALSE(cluster_tracker.exists());
-  EXPECT_EQ(cluster_tracker.info(), nullptr);
+  EXPECT_FALSE(cluster_tracker.threadLocalCluster().has_value());
 }
 
 TEST_F(ClusterUpdateTrackerTest, ClusterDoesExistAtConstructionTime) {
   EXPECT_CALL(cm_, getThreadLocalCluster(cluster_name_)).WillOnce(Return(&expected_));
 
   ClusterUpdateTracker cluster_tracker(cm_, cluster_name_);
 
-  EXPECT_TRUE(cluster_tracker.exists());
-  EXPECT_EQ(cluster_tracker.info(), expected_.cluster_.info_);
+  EXPECT_TRUE(cluster_tracker.threadLocalCluster().has_value());
+  EXPECT_EQ(cluster_tracker.threadLocalCluster()->get().info(), expected_.cluster_.info_);
 }
 
 TEST_F(ClusterUpdateTrackerTest, ShouldProperlyHandleUpdateCallbacks) {
   EXPECT_CALL(cm_, getThreadLocalCluster(cluster_name_)).WillOnce(Return(nullptr));
 
   ClusterUpdateTracker cluster_tracker(cm_, cluster_name_);
 
-  {
-    EXPECT_FALSE(cluster_tracker.exists());
-    EXPECT_EQ(cluster_tracker.info(), nullptr);
-  }
+  { EXPECT_FALSE(cluster_tracker.threadLocalCluster().has_value()); }
 
   {
     // Simulate addition of an irrelevant cluster.
     cluster_tracker.onClusterAddOrUpdate(irrelevant_);
 
-    EXPECT_FALSE(cluster_tracker.exists());
-    EXPECT_EQ(cluster_tracker.info(), nullptr);
+    EXPECT_FALSE(cluster_tracker.threadLocalCluster().has_value());
   }
 
   {
     // Simulate addition of the relevant cluster.
     cluster_tracker.onClusterAddOrUpdate(expected_);
 
-    EXPECT_TRUE(cluster_tracker.exists());
-    EXPECT_EQ(cluster_tracker.info(), expected_.cluster_.info_);
+    EXPECT_TRUE(cluster_tracker.threadLocalCluster().has_value());
```

Comment:
optional: arguably these should be ASSERTs, since the next line will fail if we don't early return.  
Given you're just changing function signatures your call if you want to clean it up.